### ISSUE TITLE
Vim(call):E117: Unknown function: <SNR>18_themis_vimspec_scope_4 in Vim 7.3

### ISSUE THREAD

**Issue by lambdalisue** [thinca/vim-themis#21]
An exception, `Vim(call):E117: Unknown function: <SNR>18_themis_vimspec_scope_4`, occur when `sudo add-apt-repository ppa:pi-rho/dev -y` is executed in `install` section of travis (mean Vim 7.4 is OK but Vim 7.3).

**It worked previously thus probably I introduced some version dependent codes and the changes triggered the exception**. However, the error message is not helped full thus I wonder if vim-themis can provide more user-friendly error message even in Vim 7.3 

- travis.yml: https://github.com/lambdalisue/vital-ArgumentParser/blob/v2/.travis.yml
- Result: https://travis-ci.org/lambdalisue/vital-ArgumentParser/jobs/64511338

I know vim-themis does not officially support Vim 7.3 but vital.vim also test them functions in Vim 7.3 thus I'll be happy if you consider about this little issue.

お時間があればで結構ですので、よろしくお願いいたします。

**Comment by thinca** [comment]
手元の環境の 7.3.429 でテストを動かしてみましたが、ちゃんと動きますね…。なんだろう。

**Comment by lambdalisue** [comment]
あれ‥‥一応三回繰り返して同じエラーが出たので報告したのですが‥‥うーんまずは解決してみます

**Comment by lambdalisue** [comment]
Finally I got it. It is quite complicated.

#### tl;dr
Do NOT define a script function which name start from `_` in `Describe`, `Context`, or whatever of Vimspec file.

#### Case 1: `Describe` + `Before` + `s:_xxx`
Use `Describe` and define a script function which name start from `_` in `Before` section.

```
Describe A
  Context B
    Before
      function! s:_bad()
      endfunction
    End
    It is Bad
      let s:Fn = function('s:_bad')
      Assert Equals(1, 1)
    End
  End
End
```

```
Bail out!  Error occurred in script loading (./test/Test.vimspec).
# style-vimspec: Error occurred in a compiled specfile of /home/alisue/.ghq/github.com/lambdalisue/vital-ArgumentParser/test/Test.vimspec at 26
# Vim(call):E117: Unknown function: <SNR>16_themis_vimspec_scope_1
```


#### Case 2: `Before` + `s:_xxx`
Define a script function which name start from `_` in `Before` section.

```
Context B
  Before
    function! s:_bad()
    endfunction
  End
  It is Bad
    let s:Fn = function('s:_bad')
    Assert Equals(1, 1)
  End
End
```

```
Bail out!  Error occurred in script loading (./test/Test.vimspec).
# /tmp/v0bqbPe/0 Line:12
#  12: function! s:themis_vimspec_bundles[-1].suite.T_00000() abort
# Vim(let):E684: list index out of range: 0
```

#### Case 3: `s:_xxx`
Define a script function which name start from `_` in `It` section.
```
Context B
  It is Bad
    function! s:_bad()
    endfunction
    let s:Fn = function('s:_bad')
    Assert Equals(1, 1)
  End
End
```

```
Bail out!  Error occurred in script loading (./test/Test.vimspec).
# style-vimspec: Error occurred in a compiled specfile of /home/alisue/.ghq/github.com/lambdalisue/vital-ArgumentParser/test/Test.vimspec at 14
# Vim(call):E121: Undefined variable: l:
```

**Comment by lambdalisue** [comment]
The above exceptions does not occur if I use `s:good` as a function name.
I'm using Vim 7.4.488 on Ubuntu 15.04. (I said Vim 7.3 was the key but it was not).

**Comment by thinca** [comment]
This is Vim's bug.  And, this is already fixed in Vim 7.4.549.  See also vim-jp/issues#659.
This is the specifications.  Please use the latest version of Vim or don't use the function name with the problem.

**Comment by lambdalisue** [comment]
That's why I didn't have the problem previously. I see (I'm using system vim accidentally).
